codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2DataFrame.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.util.IllegalReferenceCountException;
+
+/**
+ * The default {@link Http2DataFrame} implementation.
+ */
+public class DefaultHttp2DataFrame extends AbstractHttp2StreamFrame implements Http2DataFrame {
```

Comment:
Netty is pretty inconsistent with respect to final messages and message interfaces. I do see a general trend that if there are message interfaces then the class also tends to be final. Should I get rid of the interfaces, too? It's unclear what is best practice here.

(I am _totally_ for getting rid of the interfaces, but am more concerned with just "doing it the Netty way")